### PR TITLE
README: tidy up existing docs and add new troubleshooting doc

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. Simplify the sequence of board-specific environment-setup and programming method.
2. Add new troubleshooting document.
3. Add sentences on top README to know what information is.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: README.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ make
 Built binaries are in *$TIZENRT_BASEDIR/build/output/bin*.
 
 ## Supported Board / Emulator
+Here are supported boards and emulater list.
```
Comment (lokeshbv) [inline README.md]: please change, `emulater `--> `emulator`

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**